system-tests/cli/cli.test.js

```diff
@@ -105,4 +106,42 @@ describe('CLI', () => {
 			expect.stringContaining('Unexpected empty source'),
 		);
 	});
+
+	it('helps windows users with erroneous single quotes in their package.json', async () => {
+		const spy = jest.spyOn(os, 'platform').mockImplementation(() => 'win32');
+
+		return Promise.resolve(cli([`'${path.join(__dirname, 'stylesheet.css')}'`])).then(() => {
+			expect(console.log).toHaveBeenNthCalledWith(
+				1,
+				expect.stringContaining(`stylesheet.css'" were found.
+
+Please use escaped double quotes for a file pattern in your package.json, e.g.:
+
+	stylelint \\"file.css\\"
+`),
+			);
+
+			spy.mockRestore();
+		});
+	});
+
+	it('helps windows users with erroneous single quotes', async () => {
+		const spy = jest.spyOn(os, 'platform').mockImplementation(() => 'win32');
+
+		process.env = {};
```

Comment:
What this line for? If it's needed for test, maybe we can mock it instead? I'm not sure, but reassigning `process.env` looks dangerous and might cause unexpected issues in other tests.

Reply:
Ah, yeah. I missed this. It's needed because the lifecycle event is set in tests and before it was being replaced (using the unsafe method). I'll look at mocking this too, thank you!